rrs/tools/upgrade_history_internal.py

```diff
@@ -0,0 +1,617 @@
+# Internal script called by rrs_upgrade_history.py
+#
+# To detect package versions of the recipes the script uses the name of the recipe.
+#
+# Copyright (C) 2015, 2018 Intel Corporation
+# Authors: Anibal Limon <anibal.limon@linux.intel.com>
+#          Paul Eggleton <paul.eggleton@linux.intel.com>
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+
+import sys
+import os
+import optparse
+import logging
+import re
+from distutils.version import LooseVersion
+import git
+from datetime import datetime
+import calendar
+from email.utils import parsedate_tz
+
+sys.path.insert(0, os.path.realpath(os.path.join(os.path.dirname(__file__))))
+from common import common_setup, get_pv_type, load_recipes, \
+        get_logger, DryRunRollbackException
+
+common_setup()
+from layerindex import utils, recipeparse
+from layerindex.update_layer import split_recipe_fn
+
+
+def rfc2822_time_to_utc_datetime(ds):
+    tt = parsedate_tz(ds)
+    if tt is None:
+        return None
+    timestamp = calendar.timegm(tt) - tt[9]
+    return datetime.utcfromtimestamp(timestamp)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Naive datetime objects are treated by many datetime methods as local times, We recommend that you create aware timestamps with the `now` or `fromtimestamp` functions using the `tzinfo` to attribute to represent times in UTC. Do not use the [datetime.utcnow()](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) because it returns only a `naive` datetime object. 

[Learn more](https://docs.python.org/3/library/datetime.html#aware-and-naive-objects)